drivers/regulator/regulator_da1469x.c

```diff
@@ -0,0 +1,461 @@
+/*
+ * Copyright 2023 Jerzy Kasenebrg
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_smartbond_regulator
+
+#include <stdint.h>
+
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/logging/log.h>
+#include <DA1469xAB.h>
+#include <zephyr/dt-bindings/regulator/da1469x.h>
+
+LOG_MODULE_REGISTER(regulator_da1469x, CONFIG_REGULATOR_LOG_LEVEL);
+
+#define DCDC_REG_FIELD_MSK(dcdc, field) \
+	DCDC_DCDC_ ## dcdc ## _REG_DCDC_ ## dcdc ## _ ## field ## _Msk
+#define DCDC_REG_FIELD_POS(dcdc, field) \
+	DCDC_DCDC_ ## dcdc ## _REG_DCDC_ ## dcdc ## _ ## field ## _Pos
+
+#define DCDC_REG_GET(dcdc, field) \
+	((DCDC->DCDC_ ## dcdc ## _REG & DCDC_REG_FIELD_MSK(dcdc, field)) >> \
+	DCDC_REG_FIELD_POS(dcdc, field))
+
+#define DCDC_REG_SET(dcdc, field, val) \
+	DCDC->DCDC_ ## dcdc ## _REG = \
+	((DCDC->DCDC_ ## dcdc ## _REG & ~DCDC_REG_FIELD_MSK(dcdc, field)) | \
+	((val) << DCDC_REG_FIELD_POS(dcdc, field)))
+
+#define POWER_REG_FIELD_MSK(field) \
+	CRG_TOP_POWER_CTRL_REG_ ## field ## _Msk
+#define POWER_REG_FIELD_POS(field) \
+	CRG_TOP_POWER_CTRL_REG_ ## field ## _Pos
+
+#define POWER_REG_GET(field) \
+	((CRG_TOP->POWER_CTRL_REG & POWER_REG_FIELD_MSK(field)) >> POWER_REG_FIELD_POS(field))
+
+#define POWER_REG_SET(field, val) \
+	CRG_TOP->POWER_CTRL_REG = \
+	((CRG_TOP->POWER_CTRL_REG & ~POWER_REG_FIELD_MSK(field)) | \
+	((val) << POWER_REG_FIELD_POS(field)))
+
+static const uint32_t voltages_vdd_clamp[] = {
+	1037000, 1005000, 978000, 946000,
+	1120000, 1089000, 1058000, 1030000,
+	952000, 918000, 889000, 861000,
+	862000, 828000, 798000, 706000
+};
+
+static const uint32_t voltages_vdd[] = {
+	900000, 1000000, 1100000, 1200000
+};
+
+static const uint32_t voltages_vdd_sleep[] = {
+	750000, 800000, 850000, 900000,
+	950000, 1000000
+};
+
+static const uint32_t voltages_v14[] = {
+	1200000, 1250000, 1300000, 1350000,
+	1400000, 1450000, 1500000, 1550000
+};
+
+static const uint32_t voltages_v30[] = {
+	3000000, 3450000, 3300000, 3300000
```

Comment:
V30 rail does not longer support 3.45 voltages. `3450000` option should be removed.

Reply:
please ping me when regulator driver is ready

Reply:
> V30 rail does not longer support 3.45 voltages. `3450000` option should be removed.

![da14695_regulators](https://github.com/zephyrproject-rtos/zephyr/assets/23063648/03454f14-e198-4694-ba70-eb2d5234daf3)
Devices that I have do support 3.45V as specified in datasheet.

Reply:
@gmarull it's ready

Reply:
Maybe you have a TSMC version which is no longer available on the market. Here is an excerpt of POWER_CTRL_REG from the latest datasheet, that is version 3.3.
<img width="268" alt="V30_LEVEL" src="https://github.com/zephyrproject-rtos/zephyr/assets/13284615/a5444963-a988-4f85-9604-a8e3f2d23286">

Reply:
The same goes for the VDD sleep voltages, 0.95V and 1V are no longer supported in GF:
<img width="690" alt="GF_VDD_SLEEP_VOLTAGES" src="https://github.com/zephyrproject-rtos/zephyr/assets/13284615/414fb16c-c3ef-4457-9a1a-5d94a143abbf">
In contrast to TSMC variant:
<img width="509" alt="TSMC_VDD_SLEEP_VOLTAGES" src="https://github.com/zephyrproject-rtos/zephyr/assets/13284615/d89dec27-b4e5-44f4-b8f9-97303ae54786">
@kasjer The datasheet should reflect the GF silicon variant and not TSMC.

Reply:
TSMC voltages removed as requested

Reply:
Thanks for the update

Reply:
@gmarull I think you can take a look now